### ISSUE TITLE
POS Return error "Total payments amount can't be greater than xx.x"

### ISSUE THREAD

**Issue by m4mousa** [frappe/erpnext#37998]
### Information about bug

1. create POS Invoice with 2 items
2. make a return for created POS Invoice
3. return only one item
4. Checkout
5. message "Total payments amount can't be greater than xx.x" appears

### Module

accounts, selling

### Version

Frappe: 15
ERPNext: 15

### Installation method

manual install

**Comment by KevinFaro** [comment]
I'm also experiancing this. It's also an issue if the payment for the original order is greater than the actual value of the order. Such as when paying cash and getting change. I think this is pretty serious.

@GursheenK The error is thrown [here](https://github.com/frappe/erpnext/blob/4fc3a264f9988aca3acbdf4bdc703812e8fb434a/erpnext/accounts/doctype/pos_invoice/pos_invoice.py#L444) if this helps with your investigation.

We have a custom app, so have overridden this validator to no longer check this. We aren't seeing any issues yet, except for the default amount in the mode of payment matching the paid amount (full amount of cash given) instead of the actual value of the items being returned. This just means the cashier needs to set the payment amounts manually, which is okay, but not ideal as it leaves room for human error.

**Comment by Rebaz-balisani** [comment]
I am using latest version and facing exactly same issue, if the invoice contains 1 item but with more than 1 quantity or more than 1 item in both cases you want to return only few items or quanitities so this error shows up, i hope they fix it because it seems this issue is there for long time @Nihantra-Patel

**Comment by zahlenhexe** [comment]
I can confirm this.

My workflow is different.
This is how it worked in version-13:
- scanning the returned item
- click on the item and enter the negative quantity.
- then go to payment to enter the pos invoice number the item is being returned to.

This is not longer possible in version-15.
When I want to make the payment with negativ quantity in cart, the message appears:
"Incorrect value: Grand Total (Company Currency) must be >= 0.0"
so it is not possible for me to accept returns at all.

I managed to solve this in my custom pos with editing pos_controller.js: 
https://github.com/frappe/erpnext/blob/816aae968e66e1dfcc9c2c853c76688f3a65fb75/erpnext/selling/page/point_of_sale/pos_controller.js#L283

to 
```
				//checkout: () => this.save_and_checkout(),
				checkout: () => this.payment.checkout(),
```

this is the old behavior. 

I also described this in 2021 in the [forum](https://discuss.frappe.io/t/how-are-returns-of-items-made-in-pos/75486/2?u=zahlenhexe)
